sklearn/datasets/value_dropper.py

```diff
@@ -0,0 +1,295 @@
+# Author : Raghav RV <rvraghav93@gmail.com>
+#
+# Licence : BSD 3 clause
+
+import numpy as np
+import numbers
+
+from sklearn.utils import check_array
+from sklearn.utils import check_random_state
+from sklearn.utils.multiclass import type_of_target
+
+from sklearn.base import TransformerMixin
+from sklearn.preprocessing import LabelEncoder
+
+
+__all__ = ["ValueDropper"]
+
+
+class ValueDropper(TransformerMixin):
+    """Artificially insert NMAR or MCAR missing values into data.
+
+    Where,
+
+    NMAR/MNAR - Not Missing At Random / Missing Not At Random
+        When the missingness is correlated with the class classes in the
```

Comment:
This is not very precise. What you implemented is "missingness is independent of all the features given the class", which is a very specific type of correlation. I still think this is very artificial, as more missing values mean better prediction. I don't think this data generates a useful benchmark.